smwe_rom/src/compression/lc_lz2.rs

```diff
@@ -95,27 +84,93 @@ pub fn decompress(input: &[u8]) -> Result<Vec<u8>, DecompressionError> {
                 );
                 in_it = &in_it[1..];
             }
-            Command::Repeat => {
+            REPEAT..=LONG_LENGTH => {
                 if in_it.len() >= 2 {
-                    let (bytes, rest) = in_it.split_at(2);
-                    let read_start = ((bytes[0] as usize) << 8) | (bytes[1] as usize);
-                    let read_end = read_start + length;
-                    let write_start = output.len();
-                    if read_start < output.len() {
-                        output.resize(output.len() + length, 0);
-                        output.copy_within(read_start..read_end, write_start);
-                        in_it = rest;
-                    } else {
-                        return Err(LcLz2Error::RepeatRangeOutOfBounds(read_start..read_end, output.len()).into());
+                    let bytes;
+                    (bytes, in_it) = in_it.split_at(2);
+                    let from_bytes = if revised_gfx { u16::from_le_bytes } else { u16::from_be_bytes };
+                    let read_start = usize::from(from_bytes([bytes[0], bytes[1]]));
+                    if read_start >= output.len() {
+                        return Err(
+                            LcLz2Error::RepeatRangeOutOfBounds(read_start..read_start + length, output.len()).into()
+                        );
+                    }
+                    output.reserve(length);
```

Comment:
Even though this code will run just fine, I'd prefer lines 98-104 to be wrapped in an `else` block for slightly increased readability.